### ISSUE TITLE
Indent lines replaces real characters in floating windows

### ISSUE THREAD

**Issue by Terseus** [lukas-reineke/indent-blankline.nvim#110]
# 1. Observed behavior

Indent lines appears in floating windows, even replaces other characters printed by the floating window if they're in the same position the indent line has to be.

I found this with the command `:LspInfo`:

![indent-blankline-bug-floating-window-01](https://user-images.githubusercontent.com/1707139/118855862-91967180-b8d6-11eb-9707-4a53ad17a406.png)

Under "Client: efm (id 2)" it's supposed to be the text "root", not "oot" (without "r").

# 2. Expected behavior

The indent lines shouldn't be present in unrelated floating windows, or if not possible at least shouldn't replace existing characters.

This is the same floating window in the same file with the indent-blankline plugin disabled; this is what I would expect:

![indent-blankline-bug-floating-window-02](https://user-images.githubusercontent.com/1707139/118856053-cb677800-b8d6-11eb-81b2-90409924c102.png)

# 3. How to reproduce

1. Install [efm-langserver](https://github.com/mattn/efm-langserver)
2. Install the plugins [vim-plug](https://github.com/junegunn/vim-plug), [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) and indent-blankline.
3. Use the following `init.vim`
```vim
set runtimepath^=~/.vim runtimepath+=~/.vim/after
call plug#begin('~/.vim/bundle')
Plug 'lukas-reineke/indent-blankline.nvim', {'branch': 'lua'}
Plug 'neovim/nvim-lspconfig'
call plug#end()
lua << EOF
local nvim_lsp = require('lspconfig')
nvim_lsp.efm.setup {}
EOF
```
4. Open a code file.
5. Execute the command `:LspInfo`

# 4. Additional notes

System info:
* Neovim version: NVIM v0.5.0-dev+1318-g61aefaf29
* Plugin HEAD commit: ec5816267caa5fdde7a5e5a8359b9b19bc8ecf49 (the `lua` branch)

**Comment by lukas-reineke** [comment]
> The indent lines shouldn't be present in unrelated floating windows

This is hard to define.
I totally agree, it shouldn't be in the LspInfo window. But should it be disabled in all floating windows? Not sure, probably not.

The underlying reason why the indent guides overwrite text is because of the wired whitespace LspInfo uses.
There are first 2 space characters, then indentation with tab.
To support this I would need to take all the positions of spaces in between tab indent and calculate real indentation. But it's really unlikely that anyone uses this for actual code, so I don't think I will do that.

The easiest solution for now is just to put `lspinfo` in `indent_blankline_filetype_exclude`. I think I will make this the default.

**Comment by lukas-reineke** [comment]
Properly fixed now with #121
